pkg/decision/composite_service.go

```diff
@@ -69,13 +70,50 @@ func (s CompositeService) GetFeatureDecision(featureDecisionContext FeatureDecis
 
 	// @TODO: add errors
 	if s.notificationCenter != nil {
+		sourceInfo := map[string]string{}
+
+		if featureDecision.Source == FeatureTest {
+			sourceInfo["experimentKey"] = featureDecisionContext.Feature.Key
+			sourceInfo["variationKey"] = featureDecision.Variation.Key
+		}
+
 		featureInfo := map[string]interface{}{
-			"feature_key":     featureDecisionContext.Feature.Key,
-			"feature_enabled": false,
-			"source":          featureDecision.Source,
+			"featureKey":     featureDecisionContext.Feature.Key,
+			"featureEnabled": false,
+			"source":         featureDecision.Source,
+			"sourceInfo":     sourceInfo,
 		}
 		if featureDecision.Variation != nil {
-			featureInfo["feature_enabled"] = featureDecision.Variation.FeatureEnabled
+			featureInfo["featureEnabled"] = featureDecision.Variation.FeatureEnabled
+		}
+		variable := featureDecisionContext.Variable
+		if variable.ID != "" && variable.Key != "" {
+			featureInfo["variableKey"] = variable.Key
+			featureInfo["variableType"] = variable.Type
+
+			variableValue := variable.DefaultValue
+			if v, ok := featureDecision.Variation.Variables[variable.ID]; ok && featureDecision.Variation.FeatureEnabled {
+				variableValue = v.Value
+			}
+			var convertedValue interface{}
+			var e error
+
+			switch variable.Type {
+			case entities.String:
+				convertedValue, e = variableValue, nil
```

Comment:
Does `e` need to be explicitly set to `nil`? Also, can we skip this case and instead always set `convertedValue` to the `variableValue` and then it gets overwritten if any of the cases below match? That way we always have a string value as backup/default in case we ever encounter an unknown type. Additionally we don't need to do the `if e != nil` check and conditionally assign since it will already point to the `variableValue`

Reply:
error does not have to be initialized to nil explicitly, I just followed the pattern :) I will change it.  I will change it, but I don't think we can ever get unknown type, as we defined a custom type. We need to check for error, otherwise the converted value will be assigned to defaults (0.0, 0, false) in case of conversion problems